src/test/java/dk/kb/storage/facade/DsStorageFacadeTest.java

```diff
@@ -278,14 +263,83 @@ public void testIdStartsWithOrigin() throws Exception {
             e.printStackTrace();
             Assertions.fail("Should not fail since recordId now has origin as prefix",e);
         }
-              
+      
+    }
+    
+    
+   
+    
+    @Test
+    public void testRecordTree() throws Exception {        
+        //Datastructure is a tree with 5 nodes and depth=2. (See method for visualization)       
+        //Test 1: Load parent and test tree is correct
+        //Test 2: Load of the children at depth 1 and test tree is correct
+        //Test 3: Load of the children at depth 2 and test tree is correct
+        
+        //Setup
+        String parentId="doms.aviser:p";
+        String child1Id="doms.aviser:c1";
+        String child2Id="doms.aviser:c2";
+        String child1_1Id="doms.aviser:c1_1";
+        String child1_2Id="doms.aviser:c1_2";
+        
+        createTestDepth2Tree("doms.aviser"); // See this method for visualization of the tree.
+        
+      
+        //Test1, top load parent
+     
+        DsRecordDto record = DsStorageFacade.getRecordTree(parentId);       
+        
+        //Check it is parent we have
+        Assertions.assertEquals(parentId,record.getId());        
+        Assertions.assertTrue(record.getParent() == null);        
+      
+        //Check children loaded as records
+        Assertions.assertEquals(record.getChildren().size(), 2);               
+        Assertions.assertEquals(record.getChildren().get(0).getId(), child1Id);
+        Assertions.assertEquals(record.getChildren().get(1).getId(), child2Id);
+       
+        //Check child has parent loaded, both in ID list and as object
+        Assertions.assertEquals(record.getChildren().get(0).getParentId(), record.getId());
+        Assertions.assertEquals(record.getChildren().get(0).getParent().getId(), record.getId());
+        
+       // Check depth=2
+        Assertions.assertEquals(record.getChildren().get(0).getChildren().get(0).getId(), child1_1Id);
+        Assertions.assertEquals(record.getChildren().get(0).getChildren().get(0).getParent().getId(), child1Id);
+        
+        //Test2, load depth=1 record      
+        DsRecordDto child1 = DsStorageFacade.getRecordTree(child1Id);        
+ 
+        //Test the parent is now set
+        Assertions.assertEquals(child1.getParentId(), parentId);        
+        DsRecordDto parent = child1.getParent();      
+        Assertions.assertEquals(parentId,parent.getId());
```

Comment:
Static import of `assertEquals` to avoid all the `Assertions.`-prefixes in the code?

Reply:
Done